File: RedSheet.DbEntities/Models/DefaultSetting.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("DefaultSettings", Schema = "dbo")]
	[Serializable]
	public partial class DefaultSetting  
	{
 
	[Required]
	public decimal Cost { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int DefaultSettingId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string DefaultSettingName { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Status StatusId { get; set; } 

	public DefaultSetting()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/default-setting.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class DefaultSetting {
    constructor(defaultSetting?: DefaultSetting )  {
        let properties = [ "cost", "defaultSettingId", "defaultSettingName", "statusId",];
        for (let property of properties)
            if (defaultSetting && defaultSetting[property])
                this[property] = defaultSetting[property];
    }
 
    @required()
	cost : number =   undefined;
 
	defaultSettingId : number =   0 ;
 
    @required()
    @maxLength(200)
	defaultSettingName : string =   undefined;
 
    @range(0,2147483647)
	statusId : number =   undefined;


}
